new_members_handler.py

```diff
@@ -18,81 +17,88 @@ def __init__(self, user_id, captcha_message: types.Message, timestamp):
 
 
 class NewMembersHandler:
-    pendingUsers: List[PendingUser] = list()
+    pending_users: List[PendingUser] = list()
 
     def __init__(self, bot: Bot):
         self.bot = bot
 
     async def handle(self, message: types.Message):
-        if await self.__isBotAdmin(message.chat.id):
-            await self.__giveCaptcha(message)
+        if await self.__is_bot_admin(message.chat.id):
+            await self.__give_captcha(message)
 
-    async def __isBotAdmin(self, chat_id):
+    async def __is_bot_admin(self, chat_id):
         bot_id = (await self.bot.get_me()).id
         bot_chat_member = await self.bot.get_chat_member(chat_id, bot_id)
         return bot_chat_member.can_restrict_members
 
-    async def __giveCaptcha(self, message: types.Message):
-        if len(message.new_chat_members) > 1 or message.new_chat_members[0].is_bot:
+    async def __give_captcha(self, message: types.Message):
+        if len(message.new_chat_members) > 1 \
+                or message.new_chat_members[0].is_bot:
             return
 
         await self.bot.restrict_chat_member(message.chat.id,
                                             message.from_user.id,
                                             can_send_messages=False,
                                             can_send_media_messages=False,
                                             can_send_other_messages=False)
-        markup = self.__makeInlineKeyboard()
-        replied_message = await message.reply("Welcome, {0.first_name},\nAre you a spam bot or a human?"
-                                              .format(message.from_user), reply_markup=markup)
-        pendingUser = PendingUser(message.from_user.id,
-                                  replied_message,
+        markup = self.__make_inline_keyboard()
+        replied_message = await message.reply(
+            "<b>Welcome</b> {0},\nAre you a human or a spam bot?"
+            .format(self.__get_mention(message.from_user)),
+            reply_markup=markup, parse_mode=ParseMode.HTML, reply=False)
+        await message.delete()
```

Comment:
為何刪除信息

Reply:
*訊息

信息 sounds like "information" (for that I would use "資訊" though). 訊息 sounds more like "message", am not sure mainland mandarin.

mainland mandarin:
信息 could mean "information", "message"(?) or "SMS"
消息 message(?)

taiwanese mandarin:
資訊 information
訊息 message
簡訊 SMS

The group has the bot "Auto Delete Join Message" `@deljoinbot` as admin which delete join message already. I think this behavior could just replace bot. I think you should remove that bot. If you don't want to remove join messages then we just remove `await message.delete()` here. I'm fine with both.

The current behavior is that this bot read chat message, reply with the welcome message, deljoinbot delete join message. this bot show a reply to a "deleted message". I'm not sure if there are cases that users does not receive @ notification because the bot literally reply to a deleted join message. or deljoinbot delete it too fast that this bot not even receive the user join message. another problem is that we can't easily find the new member since the join message was deleted and the welcome/congrats message does not mention the user.

I added `__get_mention` now, so in this PR, it post a message to @ mention the user and then delete the join message. after the user pass the captcha, the "congrats" message also mention the user. so that we can still see the most recent new members (from the welcome or congrats message).

Also, I change the button text, because "human" and "I'm clearly a spam bot" looks very asymmetry. If you can think of better one we can also change to that.

Reply:
ahh i see thanks

Reply:
` I'm fine with both`
just make it configurable then, add a new option in config.py
`DELETE_WELCOME_MESSAGE = True`

Reply:
`The group has the bot "Auto Delete Join Message" @deljoinbot `
just checked, Micheal added him. not me. So yeah we can kick the bot and use ours

Reply:
Ok, will do this along in the PR which add checks about if the user is in one of the chats.

Reply:
> DELETE_WELCOME_MESSAGE = True

*DELETE_JOIN_MESSAGE